machine/src/output.rs

```diff
@@ -0,0 +1,32 @@
+use edfsm::{Drain, Init};
+use std::vec::Vec;
+
+#[derive(Debug)]
+pub struct OutputBuffer<A>(pub Vec<A>);
```

Comment:
I think we should also have a noop type that implements drain and init for those situations where there is no output on an FSM. This is the case for our KV store types where we are generally interested in just the events emitted, and don't need the output.